### PR TITLE
Clean up and fix types.go

### LINKED ISSUE #2530: description and properties are missing in swagger JSON
/kind bug

**What happened**:

`description` and `properties` of `VirtualMachine`, `VirtualMachineInstance`, etc. are missing in swagger JSON

Only this info is available for `VirtualMachine`

```json
"io.kubevirt.v1alpha3.VirtualMachine": {
  "type": "object",
  "x-kubernetes-group-version-kind": [
    {
      "group": "kubevirt.io",
      "kind": "VirtualMachine",
      "version": "v1alpha3"
    }
  ]
},
```

compared to `VirtualMachineList`

```json
"io.kubevirt.v1alpha3.VirtualMachineList": {
  "description": "VirtualMachineList is a list of VirtualMachine",
  "required": [
    "items"
  ],
  "properties": {
    "apiVersion": {
      "description": "APIVersion defines the versioned schema of this representation of an object. Servers should convert recognized schemas to the latest interna       l value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources",
      "type": "string"
```

**What you expected to happen**:

Properties and description should be available.

**How to reproduce it (as minimally and precisely as possible)**:
```
curl localhost:8001/openapi/v2
```

**Anything else we need to know?**:

affects openshift/kubevirt UI: https://bugzilla.redhat.com/show_bug.cgi?id=1732329

**Environment**:
- KubeVirt version (use `virtctl version`):

```
Client Version: version.Info{GitVersion:"v0.17.4", GitCommit:"adfdb8c07830b99fc79d2fd1d004e862ef70979e", GitTreeState:"clean", BuildDate:"2019-06-27T14:07:33Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{GitVersion:"v0.17.4", GitCommit:"adfdb8c07830b99fc79d2fd1d004e862ef70979e", GitTreeState:"clean", BuildDate:"2019-06-27T14:08:17Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
```

- Kubernetes version (use `kubectl version`):

```
Client Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.4+509916ce1", GitCommit:"509916ce1", GitTreeState:"clean", BuildDate:"2019-03-28T17:17:29Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.4+0ba401e", GitCommit:"0ba401e", GitTreeState:"clean", BuildDate:"2019-03-31T22:28:12Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by slintes**
**What this PR does / why we need it**:
Clean up and fix types.go:
- aligned all TypeMeta / ObjectMeta / ListMeta fields
- removed getObjectKind() / getObjectMeta() funcs
- moved type registration into registration.go, seems to be best practice

**Which issue(s) this PR fixes**:
none

**Special notes for your reviewer**:

/cc @rmohr do you remember why we added the getObjectKind() / getObjectMeta() funcs for our resources? They don't seem to be missed.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by slintes** [comment]
/hold

stupid me was looking at the wrong file, the description issue is NOT solved :/

**Comment by rmohr** [comment]
/hold 

`status` is not a required field.

**Comment by slintes** [comment]
> `status` is not a required field.

Ouch, sure, copy / paste from a bad example. Fixed

**Comment by slintes** [comment]
I found out what is causing the missing description:

a) we don't have a openapi validation schema on our CRDs (wasn't needed until now, validation is done by our apiserver/webhook)
b) even if we have, this is a feature gated alpha feature in k8s 1.14, see kubernetes/enhancements#692

@rmohr do you think this PR is worth being merged anyway? Will add the validation schema in a separate PR

**Comment by rmohr** [comment]
> /cc @rmohr do you remember why we added the getObjectKind() / getObjectMeta() funcs for our resources? They don't seem to be missed.

Maybe the client-go version which we use now does not require it anymore. But they are/were part of an interface which client-go required in order  to allow proper registration of our types.

File: staging/src/kubevirt.io/client-go/api/v1/types.go [hunk 1]
```diff
@@ -941,11 +770,11 @@ func (vl *VirtualMachineInstancePresetList) GetListMeta() meta.List {
...
 // +k8s:openapi-gen=true
 type VirtualMachine struct {
-	metav1.TypeMeta   `json:",inline"`
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
 	metav1.ObjectMeta `json:"metadata,omitempty"`
-
 	// Spec contains the specification of VirtualMachineInstance created
-	Spec VirtualMachineSpec `json:"spec,omitempty"`
+	Spec VirtualMachineSpec `json:"spec" valid:"required`
```
Comment (rmohr) [inline staging/src/kubevirt.io/client-go/api/v1/types.go]: @slintes this line is causing the CI errors.

**Comment by rmohr** [comment]
> @rmohr do you think this PR is worth being merged anyway? Will add the validation schema in a separate PR

Yes let's merge it when it is ready, seems to fix the spec.

File: staging/src/kubevirt.io/client-go/api/v1/types.go (hunk 1 above)
Comment (slintes) [reply staging/src/kubevirt.io/client-go/api/v1/types.go]: oh thanks, before this it was a merge conflict on .gitignore (which I reverted)

**Comment by slintes** [comment]
os-3.11.0-multus fails kubevirt deployment... :thinking: 

/retest

**Comment by rmohr** [comment]
We should also dump logs there. Probably worth an improvement of our log collection.

**Comment by slintes** [comment]
> We should also dump logs there. Probably worth an improvement of our log collection.

We already do that (not to the artifacts yet though). But there was no operator pod spawned yet :/

```
NAME                                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/cdi-http-import-server   1         0         0            0           6m
deployment.apps/virt-operator            2         0         0            0           6m
```

**Comment by slintes** [comment]
rebased and ready for another review
if this goes in before https://github.com/kubevirt/kubevirt/pull/2841, the latter can be closed

**Comment by slintes** [comment]
and rebased again because #2841 was merged :)

**Pull request merged**